workflow/controller/operator.go

```diff
@@ -1450,11 +1491,21 @@ func (woc *wfOperationCtx) executeResource(nodeName string, tmpl *wfv1.Template,
 		},
 		Env: execEnvVars,
 	}
+	var messages []string
 	_, err := woc.createWorkflowPod(nodeName, mainCtr, tmpl)
 	if err != nil {
-		return woc.initializeNode(nodeName, wfv1.NodeTypePod, tmpl.Name, boundaryID, wfv1.NodeError, err.Error())
+		if strings.Contains(err.Error(), exceededQuota) {
+			messages = append(messages, exceededQuota)
+			woc.log.Infof("Failed to create pod due to a lack of resources. It will be marked pending: %v", err)
+		} else {
+			return woc.initializeNode(nodeName, wfv1.NodeTypePod, tmpl.Name, boundaryID, wfv1.NodeError, err.Error())
+		}
+	}
+	if skipNodeInitialization {
```

Comment:
this branch is only reached when a node exists but the pod failed to create because of an `exceededQuota` error. Should a sleep be added here to avoid slamming the master with Pod create requests?